src/ngx_pagespeed.cc

```diff
@@ -1988,9 +1981,22 @@ void ps_send_to_pagespeed(ngx_http_request_t* r,
                           ngx_chain_t* in) {
   ngx_chain_t* cur;
   int last_buf = 0;
+
   for (cur = in; cur != NULL; cur = cur->next) {
     last_buf = cur->buf->last_buf;
 
+    if (cur->buf->flush && ctx->follow_flushes) {
+      // Calling ctx->proxy_fetch->Flush(cfg_s->handler) will be a no-op here,
+      // unless we have follow_flushes or flush_html enabled. Note that PSOL
+      // might aggregate multiple flushes into 1, and actually flush a little bit
+      // later due to html parser state and earlier scheduled operations.
+      // Also, unless we also set the flush flag on the nginx buffers we won't
+      // actually flush.
```

Comment:
> unless we also set the flush flag on the nginx buffers we won't actually flush

You added the option to do this when we build the buffer chain, so do we actually flush?